toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -454,7 +454,7 @@ func (g *PkgGraph) CreateCollapsedNode(versionedPkg *pkgjson.PackageVer, parentN
 			for _, node := range nodesToCollapse {
 				lookupErr := g.addToLookup(node, false)
 				if lookupErr != nil {
-					logger.Log.Errorf("Failed to add node (%s) back to lookup table. Error: %s", node.FriendlyName(), lookupErr)
+					err = fmt.Errorf("failed to add node (%s) back to lookup table:\n%w\n%w", node.FriendlyName(), lookupErr, err)
```

Comment:
When dealing with cleanup errors, you can either:

1. Add the cleanup error to the existing error (if any).
2. Just report the cleanup error as a warning.

Both are acceptable. However, if you are modifying existing code, it might be easier to use Option 2.